content/ec2-auto-scaling-with-multiple-instance-types-and-purchase-options/spot_resilience.md

```diff
@@ -0,0 +1,95 @@
+{{% /notice %}}
+
+
+### Increasing the application's resilience when using Spot Instances
+
+In a previous step in this workshop, you learned that the EC2 Auto Scaling group is configured to fulfill the 4 lowest-priced instance types (out of a list of 9 types) in each Availability Zone. Since Spot is spare EC2 capacity, its supply and demand vary. By diversifying your usage of capacity pools (a combination of an instance type in an Availability Zone), you increase your chances of getting the desired capacity, and decrease the potential number of interrupted instances in case Spot Instances are interrupted (when EC2 needs the capacity back for On-Demand).
+
+#### Knowledge check
+How can you increase the resilience of the web application that you deployed in this workshop, when using EC2 Spot Instances?
+
+{{%expand "Click here for the answer" %}}
+1. Add an Availability Zone - the EC2 Auto Scaling group is currently deployed in two AZs. By adding an AZ to your application, you will tap into more EC2 Spot capacity pools, further diversifying your usage and decreasing the blast radius in case a Spot interruption occurs in one of the Spot capacity pools.
+2. Add Instance Types - the 6 instance types that are configured in the Auto Scaling group have small performance variability, so it's possible to run all these instance types in a single ASG and scale on the same dynamic scaling policy. Are there any other instance types that can be added?
+3. Increase SpotInstancePools - The ASG is configured to fulfill the 4 lowest-priced instance types. Increase this number to further diversify the usage of capacity pools and decrease the blast radius.
+{{% /expand %}}
+
+#### Challenges 
+* What other Spot allocation strategy can you choose, would it be suitable for this workload? if not, when will you use it?\
+Hint: read or skim through the following [article] (https://aws.amazon.com/blogs/compute/introducing-the-capacity-optimized-allocation-strategy-for-amazon-ec2-spot-instances/)
+
+{{%expand "Click here for the answer" %}}
+If you have workloads that are not stateless and fault-tolerant like the Web application that you deployed in this workshop, you can use the capacity-optimized allocation strategy, in order to instruct the ASG to launch instances in the capacity pools which are least likely to be interrupted.
+{{% /expand %}}
+
+* By default, a Target Group linked to an Application Load Balancer distributes the requests across its registered instances using a Round Robin load balancing algorithm. Is there anything you could do to spread the load more evenly if you have backend instances from different instance families that may have slight differences on processing power? Take a look at [this](https://aws.amazon.com/about-aws/whats-new/2019/11/application-load-balancer-now-supports-least-outstanding-requests-algorithm-for-load-balancing-requests/) article.
```

Comment:
+1 for this entry
Consider changing `spread the load more evenly` for `spread the load more efficiently`; It could be said that the load is already being spread evenly (round robin) but ends up not being efficiently processed given instance performance is heterogeneous .

Reply:
Good call-out. I'll update it.